### DISCUSSION TITLE
.NET Framework Web API problems after upgrading

### DISCUSSION THREAD

**Discussion by Woudjee** [dotnet/aspnetcore#53401] [General]
Hi, for the full story, please refer to https://github.com/dotnet/runtime/issues/96980.

In short: I have a .NET Framework Web API that I have upgraded to a `Microsoft.NET.Sdk.Web` project. The reason I needed to do this was to cope with some rather tricky errors. I think that I solved these, but my application does not run anymore. I get an error saying that my project does not contain a static main method. This is correct, because as far as my knowledge goes, this only applies to .NET Core applications. I have this bit of code as my starting code:

```
public class WebApiApplication : HttpApplication
{
    protected void Application_Start()
    {
        GlobalConfiguration.Configure(Register);
    }

    private static void Register(HttpConfiguration config)
    {
        // Web API routes
        config.MapHttpAttributeRoutes();

        config.Routes.MapHttpRoute(
            name: "DefaultApi",
            routeTemplate: "api/{controller}/{action}/{id}",
            defaults: new { id = RouteParameter.Optional }
        );
    }
}
```

Any ideas why I am getting the error saying that I need a static main method? And any ideas how I can solve this? For my project it is unfortunately important that it is a .NET Framework API (not Core).

**Comment by lextm** [comment]
The suggestion that asked you to upgrade to SDK style project format was incorrect. You need to roll back those changes.

See [my comment there](https://github.com/dotnet/runtime/issues/96980#issuecomment-2027935185).
